File: tools/source/sdks/lua-sdk/examples/task_pull_once.lua
local Client = require("../src/client")

local client = Client.new("http://127.0.0.1:8080", "dev-key")
print(client:pending_tasks_request("game_client", 1).path)
print(client:claim_runtime_task_payload("task-1", "game_client", "lua-sdk-example"))
print(client:start_runtime_task_payload("task-1", "lease-token"))

File: tools/source/sdks/cpp-sdk/examples/task_pull_once.cpp
#include <iostream>
#include "../src/game_agent_engine_client.hpp"

int main() {
    GameAgentEngineClient client("http://127.0.0.1:8080", "dev-key");
    std::cout << client.pendingTasksPath("game_client", 1) << std::endl;
    std::cout << client.claimRuntimeTaskPayload("task-1", "game_client", "cpp-sdk-example") << std::endl;
    std::cout << client.startRuntimeTaskPayload("task-1", "lease-token") << std::endl;
    return 0;
}

